### ISSUE TITLE
BufferBlock.OutputAvailableAsync does not set CancellationToken when canceled

### ISSUE THREAD

**Issue by devsko** [dotnet/corefx#25635]
When I await BufferBlock<T>.OutputAvailableAsync(token) and the token gets canceled a TaskCanceledException is thrown with a CancellationToken set to CancellationToken.None.
I think this is because OutputAvailableAsyncTarget<T>.CancelAndUnlink() calls thisTarget.TrySetCanceled() without the token as parameter.

**Comment by stephentoub** [comment]
> I think this is because OutputAvailableAsyncTarget.CancelAndUnlink() calls thisTarget.TrySetCanceled() without the token as parameter.

Yes.  The library was written before the `TrySetCanceled(CancellationToken)` method was available.
